core/domain/question_services.py

```diff
@@ -304,3 +325,30 @@ def get_questions_batch(
 
     questions_batch = get_questions_by_ids(random_question_ids)
     return questions_batch
+
+
+def get_question_summaries_for_collection(collection_id):
+    """Gets a list of question summaries for a collection.
+
+    Args:
+        collection_id: str. ID of the collection.
+
+    Returns:
+        list(QuestionSummary). A list of Question Summary objects.
+    """
+    collection = collection_services.get_collection_by_id(collection_id)
+    skill_ids_of_question = {}
```

Comment:
This is confusing, it suggests that only one question is being talked about. Perhaps call it questions_to_skill_ids?

Also consider using collections.defaultdict(list). Then you might not need lines 343-344.

Reply:
updated.